### PR TITLE
Fix Sparkle group modifiers

### PR DESCRIPTION
**Pull request by flvndvd**
## Description

This PR addresses the issue with group modifiers in Sparkle. Group modifiers are intended to style an element based on the state of a parent element. However, the current implementation struggles with nested group modifiers, as the state of the highest parent affects the entire component.

For example, in the `DataTable` component, which uses a group modifier, any component rendered within it, such as a `DropdownMenu`, will have its styles altered. This poses a problem because disabling an item in the dropdown menu is overridden by the `DataTable` parent group state. These group modifiers can be prefixed. See the [documentation](https://tailwindcss.com/docs/hover-focus-and-other-states#differentiating-nested-groups) for more details.

### CODE REVIEW

**Comment by flvndvd** [comment]
cc @Duncid this might be of your interest for your next Sparkle components.

**TL;DR:** Don't use the `group` modifier without a prefix 🙏.

**Review by JulesBelveze — COMMENTED** [review]
Interesting! 
Question: not sure why in the DataTable storybook the enabled dropdown menu behaves weirdly on hover?

Screenshot 2024-09-05 at 16 15 21

**Comment by flvndvd** [comment]
Ah yeah it's my debug, let me remove it.

**Review by JulesBelveze — APPROVED** [review]
LGTM
